src/main/java/pan/Parser.java

```diff
@@ -0,0 +1,96 @@
+package pan;
+
+import java.util.Scanner;
+
+import pan.enums.Commands;
+import pan.enums.TaskStatus;
+import pan.exceptions.InternalTestCases;
+
+/**
+ * Parser - Encapsulates logic with I/O operations in the CLI
+ * @author Jerome Goh
+ */
+public class Parser {
+    private Ui ui;
+    private Scanner scanner;
+    private TaskList taskList;
+
+    /**
+     * Constructs a Parser instance.
+     *
+     * @param ui Ui instance that parser depends on.
+     * @param scanner Scanner instance responsible to deal with I/O operations in CLI.
+     * @param taskList TaskList instance responsible for CRUD operations with chatbot's Tasks.
+     */
+    public Parser(Ui ui, Scanner scanner, TaskList taskList) {
+        this.ui = ui;
+        this.scanner = scanner;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Represents the driver method to parse and validate user logic.
+     */
+    public void parseInput() {
```

Comment:
It is good that even though the  if else is long, but there is no nested if else